### PR TITLE
pkp/pkp-lib#4622 Enabling Laravel's queue worker at register_shutdown_function

### PR DESCRIPTION
**Pull request by henriqueramos**

### CODE REVIEW

File: dbscripts/xml/install.xml [hunk 1]
```diff
@@ -23,6 +23,7 @@
 	<migration class="lib.pkp.classes.migration.FilesMigration" />
 	<migration class="lib.pkp.classes.migration.GenresMigration" />
 	<migration class="lib.pkp.classes.migration.JobsMigration" />
+	<migration class="lib.pkp.classes.migration.FailedJobsMigration" />
```
Comment (henriqueramos) [inline dbscripts/xml/install.xml:26]: With this, if a job failed, he will be persisted into the failed_jobs table. Helping us to tracking down failed jobs.

**Comment by asmecher** [comment]
It still looks like test builds are failing -- the latest appears to be this one? https://travis-ci.com/github/pkp/ojs/builds/230232548

File: config.TEMPLATE.inc.php [hunk 2]
```diff
@@ -505,3 +505,17 @@
...
+;;;;;;;;;;;;;;;;;;;
+; Queues Settings ;
+;;;;;;;;;;;;;;;;;;;
+
+[queues]
+
+; Default queue driver to be used
+default_connection = "database"
+; Default queue to be used if someone forgot to associate a queue during Queue::push();
+default_queue = "queue"
```
Comment (asmecher) [inline config.TEMPLATE.inc.php:519]: If a user doesn't remember to add this entry to their configuration file, `default_queue` will be `null`. I'd recommend providing a sensible default in calls to `Config::getVar`, as you've done for `default_connection` already. (Also applies to OMP and OPS.)

File: dbscripts/xml/install.xml [hunk 3]
```diff
@@ -18,6 +18,7 @@
 	<!-- PKP schema components -->
 	<migration class="PKP\migration\install\CommonMigration" />
 	<migration class="PKP\migration\install\ControlledVocabMigration" />
+	<migration class="PKP\migration\install\FailedJobsMigration" />
```
Comment (asmecher) [inline dbscripts/xml/install.xml:21]: This will take care of installation, but the table will also need to be added during upgrade. (Also OMP and OPS)

**Review by asmecher — CHANGES_REQUESTED** [review]
A few more comments to go with https://github.com/pkp/pkp-lib/pull/7049#pullrequestreview-811755765.

File: config.TEMPLATE.inc.php (hunk 2 above)
Comment (henriqueramos) [reply config.TEMPLATE.inc.php:519]: I'm setting the default value at [BaseJob](https://github.com/pkp/pkp-lib/pull/7049/files#diff-91b0f26b34266d092022bceaf10c468644b536a8fa5b90917cf75b83f7029d88R54).

What do you think in adding "queue" as default hardcoded value on the BaseJob?

File: dbscripts/xml/upgrade.xml [hunk 4]
```diff
@@ -138,6 +138,7 @@
 		<migration class="APP\migration\upgrade\v3_4_0\I7129_IssueEntityDAORefactor" />
 		<migration class="APP\migration\upgrade\OJSv3_3_1_i6872_UrlPathMigration" />
 		<migration class="PKP\migration\upgrade\v3_4_0\I6093_AddForeignKeys" />
+		<migration class="PKP\migration\install\FailedJobsMigration" />
```
[the remark this answers is not in the document]
Comment (henriqueramos) [reply dbscripts/xml/upgrade.xml:141]: I've added a comment there.

https://github.com/pkp/pkp-lib/pull/7049/files#diff-3aebd62cc580b7dc970b0674b8e2061379d62dd29c93cd562c0f8c36d0d71361R28

**Pull request merged**